test/Fixtures.purs

```diff
@@ -125,6 +127,71 @@ txOutputBinaryFixture1 :: String
 txOutputBinaryFixture1 =
   "8258390030fb3b8539951e26f034910a5a37f22cb99d94d1d409f69ddbaea9711c12f03c1ef2e935acc35ec2e6f96c650fd3bfba3e96550504d5336100"
 
+-- | Extend this for your needs.
+type SampleTxConfig =
+  { inputs :: Array TransactionInput }
+
+-- | Build a sample transaction using convenient config
+-- | and existing one as a base.
+mkSampleTx
+  :: Transaction
+  -> (SampleTxConfig -> SampleTxConfig)
+  -> Transaction
+mkSampleTx startTx changes =
+  applyChanges startTx $ buildChanges startTx changes
+
+  where
+  buildChanges :: Transaction -> (SampleTxConfig -> SampleTxConfig) -> SampleTxConfig
+  buildChanges (Transaction { body: TxBody { inputs } }) mkChanges =
+    mkChanges { inputs }
+
+  applyChanges :: Transaction -> SampleTxConfig -> Transaction
+  applyChanges
```

Comment:
Can't we just use a record update for this? Something like `over Transaction _ { inputs = newInputs }`

Reply:
But then there's the Body newtype.

Reply:
And it will be more painful while extending SampleTxConfig

Reply:
```
applyChanges :: Transaction -> SampleTxConfig -> Transaction
applyChanges tx@(Transaction { body }) { inputs: newInputs } =
  over Transaction _ { body = over TxBody _ { inputs = newInputs } body } tx
 ```
 or
 ```
 applyChanges :: Transaction -> SampleTxConfig -> Transaction
applyChanges (Transaction tx@{ body: TxBody txBody }) { inputs: newInputs } =
  Transaction tx { body = TxBody txBody { inputs = newInputs } }
  ```
 or I think the nicest, using a bunch of lenses from Types.Transaction

 ```applyChanges tx { inputs: newInputs } = tx # _body <<< _inputs .~ newInputs```

Reply:
If there's reason to keep it for generalising SampleTxConfig, then I reckon it's fine either way.